### PR TITLE
ASK: Added shiftedOutputText attr, auto label them; Sinhala: Added

### PR DESCRIPTION
**Pull request by martholomew**
Sorry, there are a couple of changes here:

Keyboard, PointerTracker: Added the attribute shiftedOutputText to be able to output multi-character "strings" when shifted.

AnyKeyboardViewBase: Made it so that "shiftedOutputText" keys are auto label'd so that you don't have to add a shiftedKeyLabel attribute.

AnyKeyboard: Changed mShiftCodesAlways to true so that it doesn't have any problems with [ZWNJ](https://en.wikipedia.org/wiki/Zero-width_non-joiner) and [ZWJ](https://en.wikipedia.org/wiki/Zero-width_joiner) keys which are used in Indic languages.

Sinhala: Added the language as it uses the new features.

### CODE REVIEW

**Comment by martholomew** [comment]
There is one hanging issue. Since shiftedOutputText uses the onText function in AnySoftKeyboardSuggestions, suggestions are restarted whenever one is entered. I'm not sure if I should look at changing the functionality or if I should make a seperate function. Since this is a bit of an edge case, I'm sure this can be merged without worrying about it, as this also occurs [here](https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/master/addons/languages/hebrew/pack/src/main/res/xml/heb_qwerty_niqqud_shin.xml).

File: ime/app/src/main/java/com/anysoftkeyboard/keyboards/AnyKeyboard.java [hunk 1]
```diff
@@ -833,12 +833,7 @@ public AnyKey(
             // if the shift-character is a symbol, we only show it if the SHIFT is pressed,
             // not if the shift is active.
             // Note: above doesn't work, it only shows the symbols if SHIFT is being held.
-            mShiftCodesAlways =
-                    mShiftedCodes.length == 0
-                            || Character.isLetter(mShiftedCodes[0])
-                            || Character.getType(mShiftedCodes[0]) == Character.NON_SPACING_MARK
-                            || Character.getType(mShiftedCodes[0])
-                                    == Character.COMBINING_SPACING_MARK;
+            mShiftCodesAlways = true;
```
Comment (menny) [inline ime/app/src/main/java/com/anysoftkeyboard/keyboards/AnyKeyboard.java]: does this mean that `mShiftCodesAlways` is `true` is all cases? Can we remove it and all the branches that use it?

Comment (martholomew) [reply ime/app/src/main/java/com/anysoftkeyboard/keyboards/AnyKeyboard.java]: Sounds about right.

**Comment by martholomew** [comment]
I don't think we need new unit tests.

**Comment by menny** [comment]
Good. Then please remove that field and related code. This is great - I love code deletion

**Comment by menny** [comment]
thanks!

**Comment by martholomew** [comment]
Awesome, no prob!

**Pull request merged**